modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -1982,6 +1989,27 @@ public QueryMemoryManager memoryManager() {
             U.warn(log, "Custom H2 serialization is already configured, will override.");
 
         JdbcUtils.serializer = h2Serializer();
+
+        cleanSpillDirectory();
+    }
+
+    /**
+     * Cleans spill directory. Spill directory is used for disk
+     * offloading of the intermediate results of heavy queries.
+     */
+    private void cleanSpillDirectory() {
+        try {
+            File spillDir = U.resolveWorkDirectory(
+                ctx.config().getWorkDirectory(),
+                DISK_SPILL_DIR,
+                false);
+
+            U.delete(spillDir);
```

Comment:
I remember we discussed multiple nodes sharing same work directory case. Will it work as expected in such case? Should not we add a describing comment?

Reply:
`U.delete(spillDir)` tries to remove all files and directories from spillDir. If it is not succeed (because file is being used by somebody), it ignores IOException and stops removing the files. So, it cleans as much outdated files as possible.